src/hazelcore/dynamics/Elaborator_Exp.re

```diff
@@ -1,452 +1,3 @@
-  | FreeVar(_, _, _, _) => Indet
-  | InvalidText(_) => Indet
-  | Keyword(_, _, _, _) => Indet
-  | Let(_, _, _) => Indet
-  | FixF(_, _, _) => DoesNotMatch
-  | Lam(_, _, _) => DoesNotMatch
-  | Ap(_, _) => Indet
-  | BinBoolOp(_, _, _)
-  | BinIntOp(_, _, _)
-  | BinFloatOp(_, _, _)
-  | BoolLit(_) => DoesNotMatch
-  | IntLit(_) => DoesNotMatch
-  | FloatLit(_) => DoesNotMatch
-  | Inj(_, _, _) => DoesNotMatch
-  | ListNil(_) => DoesNotMatch
-  | Cons(_, _) => DoesNotMatch
-  | Triv => DoesNotMatch
-  | ConsistentCase(_)
-  | InconsistentBranches(_) => Indet
-  | EmptyHole(_, _, _) => Indet
-  | NonEmptyHole(_, _, _, _, _) => Indet
-  | FailedCast(_, _, _) => Indet
-  | InvalidOperation(_) => Indet
-  }
-and matches_cast_Cons =
-    (
-      dp1: DHPat.t,
-      dp2: DHPat.t,
-      d: DHExp.t,
-      elt_casts: list((HTyp.t, HTyp.t)),
-    )
-    : match_result =>
-  switch (d) {
-  | Cons(d1, d2) =>
-    switch (matches(dp1, DHExp.apply_casts(d1, elt_casts))) {
-    | DoesNotMatch => DoesNotMatch
-    | Indet =>
-      let list_casts =
-        List.map(
-          (c: (HTyp.t, HTyp.t)) => {
-            let (ty1, ty2) = c;
-            (HTyp.List(ty1), HTyp.List(ty2));
-          },
-          elt_casts,
-        );
-      switch (matches(dp2, DHExp.apply_casts(d2, list_casts))) {
-      | DoesNotMatch => DoesNotMatch
-      | Indet
-      | Matches(_) => Indet
-      };
-    | Matches(env1) =>
-      let list_casts =
-        List.map(
-          (c: (HTyp.t, HTyp.t)) => {
-            let (ty1, ty2) = c;
-            (HTyp.List(ty1), HTyp.List(ty2));
-          },
-          elt_casts,
-        );
-      switch (matches(dp2, DHExp.apply_casts(d2, list_casts))) {
-      | DoesNotMatch => DoesNotMatch
-      | Indet => Indet
-      | Matches(env2) => Matches(Environment.union(env1, env2))
-      };
-    }
-  | Cast(d', List(ty1), List(ty2)) =>
-    matches_cast_Cons(dp1, dp2, d', [(ty1, ty2), ...elt_casts])
-  | Cast(d', List(_), Hole) => matches_cast_Cons(dp1, dp2, d', elt_casts)
-  | Cast(d', Hole, List(_)) => matches_cast_Cons(dp1, dp2, d', elt_casts)
-  | Cast(_, _, _) => DoesNotMatch
-  | BoundVar(_) => DoesNotMatch
-  | FreeVar(_, _, _, _) => Indet
-  | InvalidText(_) => Indet
-  | Keyword(_, _, _, _) => Indet
-  | Let(_, _, _) => Indet
-  | FixF(_, _, _) => DoesNotMatch
-  | Lam(_, _, _) => DoesNotMatch
-  | Ap(_, _) => Indet
-  | BinBoolOp(_, _, _)
-  | BinIntOp(_, _, _)
-  | BinFloatOp(_, _, _)
-  | BoolLit(_) => DoesNotMatch
-  | IntLit(_) => DoesNotMatch
-  | FloatLit(_) => DoesNotMatch
-  | Inj(_, _, _) => DoesNotMatch
-  | ListNil(_) => DoesNotMatch
-  | Pair(_, _) => DoesNotMatch
-  | Triv => DoesNotMatch
-  | ConsistentCase(_)
-  | InconsistentBranches(_) => Indet
-  | EmptyHole(_, _, _) => Indet
-  | NonEmptyHole(_, _, _, _, _) => Indet
-  | FailedCast(_, _, _) => Indet
-  | InvalidOperation(_) => Indet
-  };
-
 type elab_result_lines =
   | LinesExpand(DHExp.t => DHExp.t, Contexts.t, Delta.t)
```

Comment:
rename to LinesElaborate / DoNotElaborate

there may be some other uses of Expand in hazelcore/dynamics that need the same treatment